src/components/svgs/nodes/mrt-num-line-badge.tsx

```diff
@@ -0,0 +1,130 @@
+import { RmgFields, RmgFieldsField } from '@railmapgen/rmg-components';
+import { CityCode, MonoColour } from '@railmapgen/rmg-palette-resources';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { AttrsProps } from '../../../constants/constants';
+import { MiscNodeType, Node, NodeComponentProps } from '../../../constants/nodes';
+import { AttributesWithColor, ColorField } from '../../panels/details/color-field';
+
+const MRTNumLineBadge = (props: NodeComponentProps<MRTNumLineBadgeAttributes>) => {
+    const { id, x, y, attrs, handlePointerDown, handlePointerMove, handlePointerUp } = props;
+    const { num = defaultMRTNumLineBadgeAttributes.num, color = defaultMRTNumLineBadgeAttributes.color } =
+        attrs ?? defaultMRTNumLineBadgeAttributes;
+
+    const onPointerDown = React.useCallback(
+        (e: React.PointerEvent<SVGElement>) => handlePointerDown(id, e),
+        [id, handlePointerDown]
+    );
+    const onPointerMove = React.useCallback(
+        (e: React.PointerEvent<SVGElement>) => handlePointerMove(id, e),
+        [id, handlePointerMove]
+    );
+    const onPointerUp = React.useCallback(
+        (e: React.PointerEvent<SVGElement>) => handlePointerUp(id, e),
+        [id, handlePointerUp]
+    );
+
+    const fgColor = color[3];
+    const bgColor = color[2];
+
+    return React.useMemo(
+        () => (
+            <g id={id} transform={`translate(${x}, ${y})`}>
+                <rect fill={bgColor} x="-6" y="-6" width="12" height="12" rx="6" ry="6" />
+                <text
+                    className="rmp-name__mrt"
+                    textAnchor="middle"
+                    x="0"
+                    y="0"
+                    width="12"
+                    height="12"
+                    fill={fgColor}
+                    fontSize="9"
+                    dominantBaseline="central"
+                    letterSpacing="-0.2"
+                >
+                    {num}
+                </text>
+                {/* Below is an overlay element that has all event hooks but can not be seen. */}
```

Comment:
If we won't connect this node with lines, we should be able to move `onPointerDown`, `onPointerMove`, `onPointerUp`, `style` to the `<g id={id} .../>`.

Reply:
<img width="679" alt="image" src="https://github.com/railmapgen/rmp/assets/59787082/427ebcac-c763-4c3c-8364-a2f1baf5e03d">

Is this okay?

Reply:
Sure

Reply:
Oh Ln56 should be removed as well as the purpose of a mask element is to provide a `stn_core_${id}` for `document.elementsFromPoint` and pointer events. The first is not needed anymore and the second is moved to the outer g element.